### ISSUE TITLE
React native iOS App Crash- After Upgrade to 0.76.1

### ISSUE THREAD

**Issue by Ajithkumar2203** [facebook/react-native#47587]
### Description

I had 2 issues

Build got crashed after the splash screen in testflight (Release version)
"react-native": "^0.75.1"
Xcode Version 16.0
Mac os version 15.1

After upgrade build got crashed in debug mode also cant able to run my app
"react-native": "^0.76.1"
Xcode Version 16.0
Mac os version 15.1

Initially i thought it was because of splash screen i removed that also facing the same crash i dont know why

but when i installed the previous version "react-native": "0.70.6", it is working fine in the dev and release version

Note i updated almost all the package to most recent version is anything releated to
architechture something

Here is my package json

"dependencies": {
        "@dynatrace/react-native-plugin": "^2.255.1",
        "@react-native-async-storage/async-storage": "^1.17.11",
        "@react-native-community/checkbox": "^0.5.14",
        "@react-native-community/masked-view": "^0.1.11",
        "@react-native-community/netinfo": "^9.3.7",
        "@react-native-community/progress-bar-android": "^1.0.5",
        "@react-native-community/progress-view": "^1.4.0-beta.0",
        "@react-native-picker/picker": "^2.4.8",
        "@react-navigation/drawer": "^6.5.3",
        "@react-navigation/native": "^6.0.16",
        "@react-navigation/stack": "^6.3.7",
        "i18next": "^22.0.8",
        "lodash": "^4.17.21",
        "matomo-tracker-react-native": "^0.3.0",
        "moment": "^2.29.4",
        "papaparse": "^5.3.2",
        "prop-types": "^15.8.1",
        "react": "18.1.0",
        "react-i18next": "^12.1.0",
        "react-native": "0.70.6",
        "react-native-app-auth": "^8.0.0",
        "react-native-blob-util": "^0.17.0",
        "react-native-date-picker": "^4.2.6",
        "react-native-device-info": "^10.3.0",
        "react-native-document-picker": "^8.1.3",
        "react-native-dropdown-picker": "^5.4.6",
        "react-native-elements": "^3.4.2",
        "react-native-fs": "^2.20.0",
        "react-native-gesture-handler": "^2.8.0",
        "react-native-image-crop-picker": "^0.39.0",
        "react-native-image-picker": "^4.10.2",
        "react-native-image-zoom-viewer": "^3.0.1",
        "react-native-linear-gradient": "^2.6.2",
        "react-native-loading-spinner-overlay": "^3.0.1",
        "react-native-localize": "^2.2.4",
        "react-native-mail": "^6.1.1",
        "react-native-network-bandwith-speed": "^3.0.0",
        "react-native-paper": "^5.0.0",
        "react-native-pdf": "^6.6.2",
        "react-native-reanimated": "^2.14.4",
        "react-native-safe-area-context": "^3.1.9",
        "react-native-screens": "^3.18.2",
        "react-native-sensitive-info": "^5.5.8",
        "react-native-spinkit": "^1.5.1",
        "react-native-splash-screen": "^3.3.0",
        "react-native-svg": "^13.6.0",
        "react-native-toast-message": "^2.1.5",
        "react-native-toggle-element": "^2.0.1",
        "react-native-vector-icons": "^9.2.0",
        "react-redux": "^8.0.5",
        "realm": "^11.3.0",
        "redux": "^4.2.0",
        "redux-observable": "^2.0.0",
        "redux-thunk": "^2.4.2",
        "save-dev": "^0.0.1-security",
        "reselect": "^5.1.1"
    },
    "devDependencies": {
        "@babel/core": "^7.12.9",
        "@babel/eslint-parser": "^7.14.5",
        "@babel/runtime": "^7.12.5",
        "@react-native-community/eslint-config": "^2.0.0",
        "babel-jest": "^26.6.3",
        "detox": "^18.20.4",
        "eslint": "^7.32.0",
        "identity-obj-proxy": "^3.0.0",
        "jest": "^26.6.3",
        "jest-circus": "^26.5.3",
        "metro-react-native-babel-preset": "0.72.3",
        "react-test-renderer": "18.1.0",
        "reactotron-react-native": "^5.0.3",
        "reactotron-redux": "^3.1.3",
        "redux-mock-store": "^1.5.4"
    },
    "jest": {
        "preset": "react-native",
        "timers": "fake",
        "testEnvironment": "jsdom",
        "moduleNameMapper": {
            ".+\\.(css|styl|less|sass|scss|png|jpg|ttf|woff|woff2)$": "identity-obj-proxy"
        },
        "setupFiles": [
            "<rootDir>/jest/Setup.js"
        ],
        "transformIgnorePatterns": [
            "node_modules/(?!(jest-)?@react-native|react-native|@react-native-community|@react-native-picker|react-native-vector-icons|react-icons|react-native-toggle-element|@dynatrace/react-native-plugin|matomo-tracker-react-native)"
        ],
        "transform": {}
    },
    "detox": {
        "testRunner": "jest",
        "runnerConfig": "e2e/config.json",
        "devices": {
            "simulator": {
                "type": "ios.simulator",
                "device": {
                    "id": "92749BE8-811E-4339-B1C9-C4DC51B16861"
                }
            }
        },
        "apps": {
            "ios.debug": {
                "type": "ios.app",
                "binaryPath": "ios/build/Build/Products/Debug-iphonesimulator/mobilecheck.app",
                "build": "xcodebuild -workspace ios/mobilecheck.xcworkspace -configuration Debug -sdk iphonesimulator14.2 -scheme mobilecheck -derivedDataPath ios/build",
                "name": "mobilecheck"
            }
        },
        "configurations": {
            "ios": {
                "device": "simulator",
                "app": "ios.debug",
                "behavior": {
                    "reinstallApp": "false"
                }
            }
        }
    }
# Here is my Pod file

require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '14.0'
install! 'cocoapods', :deterministic_uuids => false

 # Flags change depending on the env values.
 flags = get_default_flags()

target 'mobilecheck' do
  config = use_native_modules!

  # use_react_native!(:path => config[:reactNativePath])

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => true,
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => FlipperConfiguration.enabled,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )

  pod 'RNVectorIcons', :path => '../node_modules/react-native-vector-icons'

  pod 'react-native-dynatrace', :

### Steps to reproduce

I did all possible ways
deleted node modules package.lock json
podfile and pod.lock json
and reinstalled again

### React Native Version

0.76.0

### Output of `npx react-native info`

```text
System:
  OS: macOS 15.1
  CPU: (10) arm64 Apple M1 Pro
  Memory: 105.08 MB / 16.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.4.1
    path: ~/.nvm/versions/node/v22.4.1/bin/node
  Yarn: Not Found
  npm:
    version: 10.8.2
    path: ~/.nvm/versions/node/v22.4.1/bin/npm
  Watchman:
    version: 2024.10.28.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.16.2
    path: /opt/homebrew/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.0
      - iOS 18.0
      - macOS 15.0
      - tvOS 18.0
      - visionOS 2.0
      - watchOS 11.0
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.3 AI-223.8836.35.2231.10671973
  Xcode:
    version: 16.0/16A242d
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 21.0.1
    path: /usr/bin/javac
  Ruby:
    version: 2.6.10
    path: /usr/bin/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 15.1.2
    wanted: ^15.1.2
  react:
    installed: 18.3.1
    wanted: 18.3.1
  react-native:
    installed: 0.76.0
    wanted: 0.76.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: false
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: true
```

**Comment by maxckelly** [comment]
Have you found a solution for this? I'm getting the below as well. 

<img width="1467" alt="Screenshot 2024-11-14 at 12 03 56 pm" src="https://github.com/user-attachments/assets/ce221cd1-29b5-43b2-b534-2f8ff949f57b">

**Comment by Ajithkumar2203** [comment]
@maxckelly no still im facing an issue have u fixed

**Comment by maxckelly** [comment]
@Ajithkumar2203 - Nope haven't fixed as well. Whats our dependencies? I've been looking to see if its a lib causing this but unsure. Maybe if we compare we could find a common dependency. What do you think?

**Comment by mohanramCypherD** [comment]
@Ajithkumar2203 @maxckelly I'm facing the exact issue with ReactNative 0.76.2, MacOS 15.2, XCode 16.1 and Node 18.17.0 . Did you guys find any solution or workaround for this.

**Comment by maxckelly** [comment]
Unfortunately not I ended up downgrading for now till the above is fixed of a solution is found

**Comment by maxckelly** [comment]
@mohanramCypherD have you found a fix? still facing the issue, updated just about every dep

**Comment by mohanramCypherD** [comment]
No, I did update all dependencies but no luck. I've to debug further.

**Comment by 0hhanum** [comment]
@mohanramCypherD I'm also facing this issue too.

**Comment by kapilw360** [comment]
im also facing same issue. upgrading version 0.74.3 to latest (0.76) app is crashing.

**Comment by 0hhanum** [comment]
@kapilw360 
I solved the problem by checking the missing packages through connecting a physical device and verifying the console.

**Comment by kapilw360** [comment]
Actually, I took help from the React Native upgrade guide and updated everything accordingly, but the app is still crashing. I'll share the video, logs, and my package.json file for more details. It would be great if you could provide any suggestions or assistance on this. Thank you!

**Comment by cortinico** [comment]
> Note i updated almost all the package to most recent version is anything releated to
> architechture something

Have you tried enabling/disabling the New Architecture?

Also what's the stacktrace?

**Comment by cipolleschi** [comment]
It's really hard to help out without a reproducer and a stack trace.

If the crash is the same one reported here by @maxckelly: https://github.com/facebook/react-native/issues/47587#issuecomment-2475137236
This has been fixed by this commit https://github.com/facebook/react-native/commit/7c60a65aced91b5c2d5ac08846e2869b475c3b31 and it will be picked in the [0.76.4 release](https://github.com/reactwg/react-native-releases/issues/655).

For other crashes, we need more information of what's happening in your app, or we can't provide any guidance.

**Comment by Ajithkumar2203** [comment]
> Actually, I took help from the React Native upgrade guide and updated everything accordingly, but the app is still crashing. I'll share the video, logs, and my package.json file for more details. It would be great if you could provide any suggestions or assistance on this. Thank you!

Please provide,
If the issue is the same as mine, I will support you

**Comment by kapilw360** [comment]
https://github.com/user-attachments/assets/1cdfadb1-9ae1-4c51-873c-339b872c9763



and the logs is below

[Firebase/Crashlytics] Version 11.5.0
+[RNFBSharedUtils getConfigBooleanValue:key:defaultValue:] [Line 155] RNFBCrashlyticsInit crashlytics_debug_enabled via RNFBJSON: 1
+[RNFBSharedUtils getConfigBooleanValue:key:defaultValue:] [Line 165] RNFBCrashlyticsInit crashlytics_debug_enabled final value: 1
+[RNFBCrashlyticsInitProvider isCrashlyticsCollectionEnabled] [Line 79] RNFBCrashlyticsInit specific key crashlytics_auto_collection_enabled not set, falling back to general key app_data_collection_default_enabled with default 1 if it does not exist.
+[RNFBSharedUtils getConfigBooleanValue:key:defaultValue:] [Line 162] RNFBCrashlyticsInit app_data_collection_default_enabled via RNFBMeta: 1
+[RNFBSharedUtils getConfigBooleanValue:key:defaultValue:] [Line 165] RNFBCrashlyticsInit app_data_collection_default_enabled final value: 1
+[RNFBCrashlyticsInitProvider componentsToRegister]_block_invoke [Line 129] RNFBCrashlyticsInit initialization successful
11.5.0 - [FirebaseCrashlytics][I-CLS000000] Crashlytics skipped rotating the Install ID during urgent mode because it is run on the main thread, which can't succeed. This can happen if the app crashed the last run and Crashlytics is uploading urgently.
11.5.0 - [FirebaseMessaging][I-FCM001000] FIRMessaging Remote Notifications proxy enabled, will swizzle remote notification receiver handlers. If you'd prefer to manually integrate Firebase Messaging, add "FirebaseAppDelegateProxyEnabled" to your Info.plist, and set it to NO. Follow the instructions at:
https://firebase.google.com/docs/cloud-messaging/ios/client#method_swizzling_in_firebase_messaging
to ensure proper integration.
[HealthKit] Background observers added to the app
11.5.0 - [FirebaseAnalytics][I-ACS023007] Analytics v.11.5.0 started
11.5.0 - [FirebaseAnalytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
[HealthKit] Background delivery enabled for ActiveEnergyBurned
[HealthKit] Background observer set up for ActiveEnergyBurned
Sending `healthKit:ActiveEnergyBurned:setup:success` with no listeners registered.
libc++abi: terminating due to uncaught exception of type std::runtime_error: Feature flags were accessed before being overridden: fuseboxEnabledDebug

i think because of the last line its crashing

**Comment by cipolleschi** [comment]
Can I see the AppDelegate?
This error usually happens because the app is trying to access the React Native runtime before it has been initialized. So probably there are some statement in the app delegate that must be moved to a different place.

**Comment by kapilw360** [comment]
> Can I see the AppDelegate? This error usually happens because the app is trying to access the React Native runtime before it has been initialized. So probably there are some statement in the app delegate that must be moved to a different place.

yahh sure. 
#import "AppDelegate.h"
#import <Firebase.h>
#import "RCTAppleHealthKit.h"
#import <React/RCTBundleURLProvider.h>
#import <React/RCTLinkingManager.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  [FIRApp configure];
  self.moduleName = @"Wellness360";
  // You can add your custom initial props in the dictionary below.
  // They will be passed down to the ViewController used by React Native.
  self.initialProps = @{};
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self launchOptions:launchOptions];
  [[RCTAppleHealthKit new] initializeBackgroundObservers:bridge];
  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
  return [self bundleURL];
}
 
- (NSURL *)bundleURL
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

- (BOOL)application:(UIApplication *)application
   openURL:(NSURL *)url
   options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options
{
  return [RCTLinkingManager application:application openURL:url options:options];
}

@end

**Comment by cipolleschi** [comment]
yep... same problem reported in another issue. Have a look at my answer [here](https://github.com/facebook/react-native/issues/47202#issuecomment-2439063613)

**Comment by maxckelly** [comment]
Hi @cipolleschi - I just updated to "react-native": "^0.76.5" and still getting the issue. See below screenshot.

<img width="1458" alt="Screenshot 2024-12-11 at 9 48 05 am" src="https://github.com/user-attachments/assets/97eff89a-66b7-4575-b40c-085c8633f28a">

**Comment by cipolleschi** [comment]
@maxckelly if the root issue is the same, upgrading RN version is not the solution.

Can you share your AppDelegate? The problem might be that the app is trying to access React native before it is initialized.

**Comment by maxckelly** [comment]
Hey @cipolleschi 

Thank you for the response. Please see the below files

AppDelegate.mm

```
#import "AppDelegate.h"

#import <React/RCTBundleURLProvider.h>

#import "RNSplashScreen.h"
#import <RNBranch/RNBranch.h>

@implementation AppDelegate
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  self.moduleName = @"aLifeLived";  
  // You can add your custom initial props in the dictionary below.
  // They will be passed down to the ViewController used by React Native.
  self.initialProps = @{};
  
  bool didLaunchFinish = [super application:application didFinishLaunchingWithOptions:launchOptions];

    [RNSplashScreen show];
  
  // Branch start
    [RNBranch.branch checkPasteboardOnInstall];
    [RNBranch initSessionWithLaunchOptions:launchOptions isReferrable:YES];
    NSURL *jsCodeLocation;
  // Branch stop
  
  return didLaunchFinish;
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
  return [self bundleURL];
}

- (NSURL *)bundleURL
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

- (BOOL)application:(UIApplication *)app openURL:(NSURL *)url options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
  [RNBranch application:app openURL:url options:options];
  return YES;
}

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray<id<UIUserActivityRestoring>> * _Nullable))restorationHandler {
  [RNBranch continueUserActivity:userActivity];
  return YES;
}

@end
```

AppDelegate.h
```
#import <RCTAppDelegate.h>
#import <Expo/Expo.h>
#import <UIKit/UIKit.h>

@interface AppDelegate : EXAppDelegateWrapper

@end
```

**Comment by cipolleschi** [comment]
App delegate looks good.
I see that your AppDelegate extends EXAppDelegateWrapper. This suggests that the app is an Expo app.

I think you should open the issue on the Expo repository first: `github.com/expo/expo`. It could be that the issue arrives from there.

i really need a reproducer to understand what's going on. One thing we can try is to inspect the stacktrace on the left.
Can you click on line 15 
<img width="1458" alt="394486895-97eff89a-66b7-4575-b40c-085c8633f28a" src="https://github.com/user-attachments/assets/85e6bfb5-fb24-4555-9e88-dc513c95c6d9" />
and try to understand in the bottom panel which Legacy Module is trying to be provided?

**Comment by maxckelly** [comment]
Hi @cipolleschi 

Again thank you for your help its greatly appreciated. 

I've clicked on 15 and see the below screenshot. Does this indicate that React Native track player is the issue? 
<img width="559" alt="Screenshot 2024-12-23 at 11 04 08 am" src="https://github.com/user-attachments/assets/e024698d-82c0-4f5d-bb80-03d28e502e72" />

**Comment by cipolleschi** [comment]
Yes, that's likely the problem
Can you try to create a simple reproducer using this [template](https://github.com/react-native-community/reproducer-react-native) and adding only the `RNTrackPlayer` library? If it crashes, in a similar way, it means that that library is the problem. We can take it from there and we can try to improve the interop layer so that libraries with a similar setup can work out of the box.

Thank you! 🙏

**Comment by leecuong666** [comment]
I guess I also have the same issue in react-native-reanimated-carousel library. react native 0.76.5

**Comment by iiagodias** [comment]
I discovered something interesting, after removing the use of Modal in the project, the app started running again.

**Comment by cipolleschi** [comment]
@iiagodias the issue in #48065 is completely different and the problem is related to how the app is initialized.

Modals should not crash the application. The question is more: what were you doing in your modal? This crash is caused by some module running in interoperability mode that is passing a `null` string to React Native when we expect to receive a non-null string. And the app crashes.

To understand which dependency is causing the crash, you can:
1. crash the app in a debugging session in Xcode, 
2. click on the row that states `[RCTTurboModuleManager provideLegacyModule]`
3. read the `moduleName` variable in the debugger console that opens in the bottom pane.

Then, please, create a repro using this [template](https://github.com/react-native-community/reproducer-react-native) and share it with us.

**Comment by amitesh786** [comment]
@cipolleschi  I came across the thread and believe I might be encountering the same error. Could you kindly share the solution if you have one?

```
#import "AppDelegate.h"
#import <React/RCTBundleURLProvider.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  self.moduleName = @"Privacy";
  self.initialProps = @{};

 return [super application:application didFinishLaunchingWithOptions:launchOptions];
}

- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
  return [self bundleURL];
}
 
- (NSURL *)bundleURL
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}

@end
```

I’m attaching a screenshot for reference. However, the **iOS** app continues to crash on **React Native 0.76.6**.

<img width="1435" alt="Image" src="https://github.com/user-attachments/assets/6853962d-b432-4e3b-a894-b914a63a669e" />

**Comment by cipolleschi** [comment]
We don't have a specific solution right now. It would greatly help if you can create a repro using this [template](https://github.com/react-native-community/reproducer-react-native) where you just add the library that adds the  `RNBluetoothClassic` module, so we can try to run it locally and figure out what's breaking.

**Comment by amitesh786** [comment]
> We don't have a specific solution right now. It would greatly help if you can create a repro using this [template](https://github.com/react-native-community/reproducer-react-native) where you just add the library that adds the `RNBluetoothClassic` module, so we can try to run it locally and figure out what's breaking.

I cannot access GitHub in this environment due to security restrictions.

> For this specific case, `react-native-bluetooth-classic` does not support the new arch right now. The last release was more than a year ago and 6 month ago they released an RC version of the library. Have you tried to update to that version?

Currently, we are using the latest version of **"react-native-bluetooth-classic": "1.73.0-rc.12"**, which is not compatible with **React Native 0.76.6**

Unfortunately, I cannot use the provided [template](https://github.com/react-native-community/reproducer-react-native) in my current environment. Would it be acceptable if I download the zip and manually add the required configurations to reproduce the issue? Alternatively, if you have another suggested approach for creating a repro, I’d be happy to follow it.

Thanks @cipolleschi

**Comment by cipolleschi** [comment]
I investigated this issue with the RNBluetoothClassic library.

TL;DR: The app is not compatible with the interop layer.

The longer answer:
- This library defines a module in [ObjC](https://github.com/kenjdavidson/react-native-bluetooth-classic/blob/main/ios/RNBluetoothClassic.m#L30)
- The module export some methods as [external methods](https://github.com/kenjdavidson/react-native-bluetooth-classic/blob/main/ios/RNBluetoothClassic.m#L30-L31)
- These methods are implemented in a [Swift equivalent module](https://github.com/kenjdavidson/react-native-bluetooth-classic/blob/main/ios/RNBluetoothClassic.swift#L182-L187)
- The interop layer works by [asking the ObjectiveC](https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTInteropTurboModule.mm#L122-L126) runtime about the methods a specific class is exposing 
- The problem happens when the signatures between ObjectiveC and Swift mismatch. For example the [`available:` method in ObjectiveC](https://github.com/kenjdavidson/react-native-bluetooth-classic/blob/main/ios/RNBluetoothClassic.m#L134-L136) is remapped to [`availableFromDevice` in Swift](https://github.com/kenjdavidson/react-native-bluetooth-classic/blob/main/ios/RNBluetoothClassic.swift#L502-L506).
- Therefore, the runtime fails to find the method signature and it fails to parse the method (see image)

<img width="1236" alt="Image" src="https://github.com/user-attachments/assets/3e3b2bdf-8e9b-4876-8b8c-44980b8e7c10" />

I manage to "fix" this by applying this change in the RNBluetoothClassic.m file

```diff
-RCT_EXTERN_METHOD(available: (NSString *)deviceId
+RCT_EXTERN_METHOD(availableFromDevice: (NSString *)deviceId
                  resolver: (RCTPromiseResolveBlock)resolve
                  rejecter: (RCTPromiseRejectBlock)reject)
```

You can patch package the library to unblock yourself @amitesh786 .

I don't know if, right now, there is a way to map `available`in objective-C to `availableFromDevice` in Swift.
I'll have to investigate more. 

Meanwhile, it is worth opening an issue in the [`react-native-bluetooth-classic`](https://github.com/kenjdavidson/react-native-bluetooth-classic) repo to let them know about this problem.

**Comment by cipolleschi** [comment]
We mitigated the Interop Layer issue with this PR: https://github.com/facebook/react-native/pull/49072

Something worth mentioning is that the problem happens when there is a library that has some swift code and that maps the original ObjectiveC signature to a different signature in Swift. Please notice that this remapping is not working in the Old Architecture as well.

The reason why in the old architecture "is working" is that it is not crashing right away, as it does in the New Architecture, and the apps are probably not using the broken method.

**Comment by MarianAlexandruAlecu** [comment]
@cipolleschi when will https://github.com/facebook/react-native/pull/49072 be released?

**Comment by cipolleschi** [comment]
It landed today. I need to create pick requests and then the release crew will decide when to release 0.77 and 0.76. We are currently working on 0.78 already. Hopeful next week we might release 0.77.1 given we already have a bunch of relevant commits.
